pkg/apiserver/api_installer.go

```diff
@@ -689,6 +709,9 @@ var _ ScopeNamer = legacyScopeNaming{}
 
 // Namespace returns the namespace from the query or the default.
 func (n legacyScopeNaming) Namespace(req *restful.Request) (namespace string, err error) {
+	if n.scope.Name() == meta.RESTScopeNameRoot {
```

Comment:
Events are namespaced scoped, so all three combinations are valid (empty and default to "default", set, and empty and no default applied).

The larger issue which I won't fix here is that we have a few other resources that are namespaced AND root scoped - events, and in openshift the subject and resource access reviews (which you can ask "can I do this" which involves actions in and out of namespaces).  For right now, they are listed as namespaced actions and api.NamespaceAll is a valid option for them.  We probably need to more formally define resources that support NamespaceAll, since there are other actions beyond "get collection" you might want to expose on a namespace all resource (post on the root to create a root event).